clients/horizonclient/client.go

```diff
@@ -411,10 +466,32 @@ func (c *Client) SubmitTransactionXDR(transactionXdr string) (txSuccess hProtoco
 	return
 }
 
-// SubmitTransaction submits a transaction to the network. err can be either error object or horizon.Error object.
+// SubmitTransaction submits a transaction to the network. err can be either
+// error object or horizon.Error object.
+//
+// This function will always check if the destination account requires a memo in
+// the transaction (as defined in https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0029.md)
+// if you want to skip this check, use `SubmitTransactionWithOptions{tx, SubmitTxOpts{SkipMemoRequiredCheck: true})`
+//
 // See https://www.stellar.org/developers/horizon/reference/endpoints/transactions-create.html
-func (c *Client) SubmitTransaction(transaction txnbuild.Transaction) (txSuccess hProtocol.TransactionSuccess,
-	err error) {
+func (c *Client) SubmitTransaction(transaction txnbuild.Transaction) (txSuccess hProtocol.TransactionSuccess, err error) {
+	return c.SubmitTransactionWithOptions(transaction, SubmitTxOpts{SkipMemoRequiredCheck: false})
```

Comment:
Specifying the default value of fields in SubmitTxOpts is unnecessary, and if we add more options I imagine we won't keep coming back to this line to add them. Probably unnecessary now?
```suggestion
	return c.SubmitTransactionWithOptions(transaction, SubmitTxOpts{})
```

Reply:
You are right, I did it on purpose to be explicit about it. I was thinking more from the perspective of someone who is new to go and is not aware defaults. But you are right, we can update it.